api/v1/stats/sushi/PKPStatsSushiHandler.inc.php

```diff
@@ -0,0 +1,230 @@
+<?php
+
+/**
+ * @file api/v1/stats/sushi/PKPStatsSushiHandler.inc.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2003-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class PKPStatsSushiHandler
+ * @ingroup api_v1_stats
+ *
+ * @brief Handle API requests for COUNTER R5 SUSHI statistics.
+ *
+ */
+
+use APP\facades\Repo;
+use APP\sushi\PR;
+use APP\sushi\PR_P1;
+use PKP\core\APIResponse;
+use PKP\handler\APIHandler;
+use PKP\security\authorization\ContextRequiredPolicy;
+use PKP\sushi\CounterR5Report;
+use PKP\sushi\SushiException;
+use Slim\Http\Request as SlimHttpRequest;
+
+class PKPStatsSushiHandler extends APIHandler
+{
+    /**
+     * Constructor
+     */
+    public function __construct()
+    {
+        $this->_handlerPath = 'stats/sushi';
+        $this->_endpoints = [
+            'GET' => $this->getGETDefinitions()
+        ];
+        parent::__construct();
+    }
+
+    /**
+     * Get this API's endpoints definitions
+     */
+    protected function getGETDefinitions(): array
+    {
+        $roles = [];
+        return [
+            [
+                'pattern' => $this->getEndpointPattern() . '/status',
+                'handler' => [$this, 'getStatus'],
+                'roles' => $roles
+            ],
+            [
+                'pattern' => $this->getEndpointPattern() . '/members',
+                'handler' => [$this, 'getMembers'],
+                'roles' => $roles
+            ],
+            [
+                'pattern' => $this->getEndpointPattern() . '/reports',
+                'handler' => [$this, 'getReports'],
+                'roles' => $roles
+            ],
+            [
+                'pattern' => $this->getEndpointPattern() . '/reports/pr',
+                'handler' => [$this, 'getReportsPR'],
+                'roles' => $roles
+            ],
+            [
+                'pattern' => $this->getEndpointPattern() . '/reports/pr_p1',
+                'handler' => [$this, 'getReportsPR1'],
+                'roles' => $roles
+            ],
+        ];
+    }
+
+    /**
+     * @copydoc PKPHandler::authorize()
+     */
+    public function authorize($request, &$args, $roleAssignments)
```

Comment:
I think this leaves SUSHI reports open to anyone who knows the URL and has a login (e.g. any author). I would expect these endpoints to be available to managers, admins, and subscription managers. For cases where journals want to grant external access, the journal will want to create a Subscription Manager account with an API key and provide this to the consumer. (The OJS/OMP/OPS API already handles API keys via `ApiTokenDecodingMiddleware`.)

Reply:
Hmmm...
We decided:

> Yes, this is the minimal implementation for per-customer reports. We can make per-customer reports public, or, at our option, require authentication. I am not opposed to making them public by default; if a service provider want to add in authentication, this could be a contribution or sponsored development.

here: https://github.com/pkp/pkp-lib/issues/6781#issuecomment-945802002.
And to implement the opt-out for institutions.

But I could also change it to restrict the access to those roles.
Just to double check it: @ctgraham, what do you think?